src/QsCompiler/Compiler/CompilationLoader.cs

```diff
@@ -451,49 +456,68 @@ public CompilationLoader(SourceLoader loadSources, ReferenceLoader loadReference
                 else this.LogAndUpdate(ref this.CompilationStatus.Validation, ErrorCode.MissingEntryPoint, Array.Empty<string>());
             }
 
-            // executing the specified rewrite steps
-
             if (!Uri.TryCreate(Assembly.GetExecutingAssembly().CodeBase, UriKind.Absolute, out Uri thisDllUri))
             { thisDllUri = new Uri(Path.GetFullPath(".", "CompilationLoader.cs")); }
 
             if (!String.IsNullOrWhiteSpace(this.Config.TargetPackageAssembly))
             {
+                RaiseCompilationTaskStart("Build", "ReplaceTargetSpecificImplementations");
                 this.ReplaceTargetSpecificImplementations(thisDllUri, out this.CompilationOutput);
+                RaiseCompilationTaskEnd("Build", "ReplaceTargetSpecificImplementations");
             }
 
+            RaiseCompilationTaskEnd("OverallCompilation", "Build");
+
+            // executing the specified rewrite steps
+
+            var steps = new List<(int, Func<QsCompilation>)>();
+
             if (this.Config.ConvertClassicalControl)
             {
                 var rewriteStep = new RewriteSteps.LoadedStep(new ClassicallyControlled(), typeof(IRewriteStep), thisDllUri);

```

Comment:
Would it be possible to refactor these if blocks into an extension method? They all look very similar.

Reply:
Annoyingly not really, since it is not happy to have ref inside a local method. So I don't really get around the second line.